buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -583,18 +583,34 @@ private static void setSingleBytes(final AbstractByteBuf buf, final long addr, f
     static void getBytes(AbstractByteBuf buf, long addr, int index, OutputStream out, int length) throws IOException {
         buf.checkIndex(index, length);
         if (length != 0) {
-            ByteBuf tmpBuf = buf.alloc().heapBuffer(length);
-            try {
-                byte[] tmp = tmpBuf.array();
-                int offset = tmpBuf.arrayOffset();
-                PlatformDependent.copyMemory(addr, tmp, offset, length);
-                out.write(tmp, offset, length);
-            } finally {
-                tmpBuf.release();
+            int len = Math.min(length, ByteBufUtil.WRITE_CHUNK_SIZE);
```

Comment:
Can we just do: `Math.min(Math.min(length, ByteBufUtil.WRITE_CHUNK_SIZE), length)`? This would eliminate the reason to pass 2 length parameters to the `getBytes()` below.

Reply:
hmm wouldn't we still need it ? I am confused why we not need it with this change ... Can you explain

Reply:
Because I'm an idiot? Didn't notice `outLen` is updated at the end of the loop. :-)

Reply:
lol ok that explains why =P